include/dgl/immutable_graph.h

```diff
@@ -68,6 +68,14 @@ class CSR : public GraphInterface {
     LOG(FATAL) << "CSR graph does not allow mutation.";
   }
 
+  DLContext Context() const override {
+    return indptr_->ctx;
+  }
+
+  uint8_t NumBits() const override {
+    return indptr_->dtype.bits;
```

Comment:
can we use `indices_->dtype.bits`? in theory, `indptr` and `indices` can have different number of bits. we can scale to a graph with more than 4B edges and still use 32-bit indices.

Reply:
Done.